### PR TITLE
Adds icons to link

### PR DESCRIPTION
**Pull request by FinalAngel**

### CODE REVIEW

File: djangocms_bootstrap4/contrib/bootstrap4_link/constants.py [hunk 1]
```diff
@@ -15,3 +15,9 @@
     ('', _('Medium')),
     ('btn-lg', _('Large')),
 )
+
+LINK_ICONS = getattr(
```
Comment (vxsx) [inline djangocms_bootstrap4/contrib/bootstrap4_link/constants.py]: probably `USE_LINK_ICONS`

**Review by vxsx — APPROVED** [review]
looks ok

File: djangocms_bootstrap4/contrib/bootstrap4_link/templates/djangocms_bootstrap4/link/default/link.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        {% with icon_class=instance.icon_left instance=instance.icon_left %}{% include "djangocms_icon/includes/icon.html" %}{% endwith %}
+    {% endif %}
+    {% for plugin in instance.child_plugin_instances %}
+        {% render_plugin plugin %}{% empty %}{{ instance.name }}
+    {% endfor %}
+    {% if instance.icon_right %}
+        {% with icon_class=instance.icon_right instance=instance.icon_left %}{% include "djangocms_icon/includes/icon.html" %}{% endwith %}
+    {% endif %}
+</a>{% spaceless %}
+{% endspaceless %}{% comment %}
```
Comment (filwaitman) [inline djangocms_bootstrap4/contrib/bootstrap4_link/templates/djangocms_bootstrap4/link/default/link.html]: Any particular reason to keep the comment tag?

Comment (filwaitman) [reply djangocms_bootstrap4/contrib/bootstrap4_link/templates/djangocms_bootstrap4/link/default/link.html]: Also, Why do we need those `{% spaceless %}{% endspaceless %}` with no content inside them?

File: djangocms_bootstrap4/contrib/bootstrap4_link/constants.py (hunk 1 above)
Comment (filwaitman) [inline djangocms_bootstrap4/contrib/bootstrap4_link/constants.py]: Maybe a better name for this var? At first sight I thought it was a list, not a boolean.

File: djangocms_bootstrap4/contrib/bootstrap4_link/constants.py [hunk 3]
```diff
@@ -15,3 +15,9 @@
...
+    settings,
+    'DJANGOCMS_BOOTSTRAP4_ICONS',
```
Comment (filwaitman) [inline djangocms_bootstrap4/contrib/bootstrap4_link/constants.py]: Maybe a better name for this var? At first sight I thought it was a list, not a boolean.

File: djangocms_bootstrap4/contrib/bootstrap4_link/templates/djangocms_bootstrap4/link/default/link.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{% load cms_tags %}{% spaceless %}
+
+{# this needs to be in one line for rendering purpose #}
+{% endspaceless %}<a href="{{ link }}"{% if instance.target %} target="{{ instance.target }}"{% endif %}{% if instance.attributes %} {{ instance.attributes_str }}{% endif %}>
+    {% if instance.icon_left %}
```
Comment (filwaitman) [inline djangocms_bootstrap4/contrib/bootstrap4_link/templates/djangocms_bootstrap4/link/default/link.html:5]: If:
```
1) LINK_ICONS was true (past)
2) instance.icon_left is present
3) LINK_ICONS is true (present)
```
Should we display the icon?

File: djangocms_bootstrap4/contrib/bootstrap4_link/templates/djangocms_bootstrap4/link/default/link.html [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        {% with icon_class=instance.icon_left instance=instance.icon_left %}{% include "djangocms_icon/includes/icon.html" %}{% endwith %}
+    {% endif %}
+    {% for plugin in instance.child_plugin_instances %}
+        {% render_plugin plugin %}{% empty %}{{ instance.name }}
+    {% endfor %}
+    {% if instance.icon_right %}
```
Comment (filwaitman) [inline djangocms_bootstrap4/contrib/bootstrap4_link/templates/djangocms_bootstrap4/link/default/link.html:11]: If:
```
1) LINK_ICONS was true (past)
2) instance.icon_right is present
3) LINK_ICONS is true (present)
```
Should we display the icon?

**Review by filwaitman — CHANGES_REQUESTED** [review]

File: djangocms_bootstrap4/contrib/bootstrap4_link/constants.py (hunk 1 above)
Comment (filwaitman) [reply djangocms_bootstrap4/contrib/bootstrap4_link/constants.py]: @vxsx suggestion sounds good (`USE_LINK_ICONS`)

File: djangocms_bootstrap4/contrib/bootstrap4_link/constants.py (hunk 3 above)
Comment (filwaitman) [reply djangocms_bootstrap4/contrib/bootstrap4_link/constants.py]: For consistency sake I would use something like `DJANGOCMS_BOOTSTRAP4_USE_LINK_ICONS`

**Review by filwaitman — APPROVED** [review]

File: djangocms_bootstrap4/contrib/bootstrap4_link/templates/djangocms_bootstrap4/link/default/link.html [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+        {% with icon_class=instance.icon_right instance=instance.icon_right %}{% include "djangocms_icon/includes/icon.html" %}{% endwith %}
+    {% endif %}
+</a>{% spaceless %}
+{% endspaceless %}
```
Comment (vxsx) [inline djangocms_bootstrap4/contrib/bootstrap4_link/templates/djangocms_bootstrap4/link/default/link.html:15]: should be no new line at the end of file

**Pull request merged**